vars/neoDeploy.groovy

```diff
@@ -1,8 +1,19 @@
 import com.sap.piper.Utils
+
+import groovy.transform.Field
+
 import com.sap.piper.ConfigurationLoader
 import com.sap.piper.ConfigurationMerger
 import com.sap.piper.ConfigurationType
 
+//
+// envProps may be overwritten by tests, but only by tests.
+// [Q] Why not simply using the Map returned by getenv() itself?
+// [A] The unmodifiable map returned by getenv() is not serializable
+//     Since everythings needs to be serializabe (CPS pattern) we
+//     cannot use that map directly.
+@Field Map envProps = System.getenv().findAll { e -> true }
```

Comment:
Why `e -> `? Shouldn't only `true` do the job as well?

Reply:
> can we be sure findAll will always (in further releases) return a modifiable map.

In case this behavior is changed we will get a test failure, so we will find it at the earliest. It is not documented explicitly that the map is mutable. But since there is an ```asImmutable``` I would expect always to get mutables (ok, this is not a rock-solide prove).

BTW: the point here is not immutability. The map is immutable, but the point is serializabiliity. The map returned by ```getenv()``` is not _serializable_, of course this map is also not modifiable.

Of course I can change to good old style 
```
Map envProps = [:]
envProps.putAll(System.getenv())
```

Reply:
with the comments to explain it is a 4-liner 😉 
You can decide, both is ok.

Reply:
But the comment makes sense regardless if we use ```findAll```or a more old-school way ...
The comment would only make no sense in case we could use the map returned by ```getEnv()``` directly, which is not possible.